wisdem/rotorse/rotor_power.py

```diff
@@ -630,51 +631,52 @@ def constr_Tmax(pitch_i, Uhub_i, Omega_rpm_i):
             return 1e-5 * (max_T - float(myout["T"]))
 
         # Maximize power until rated
-        for i in range(i_3):
-            # No need to optimize if already doing well
-            if (
-                ((Omega[i] == Omega_tsr[i]) and not self.peak_thrust_shaving)
-                or ((Omega[i] == Omega_tsr[i]) and self.peak_thrust_shaving and (T[i] <= max_T))
-                or (found_rated and (i == i_rated))
-            ):
-                continue
-
-            # Find pitch value that gives highest power rating
-            pitch0 = pitch[i] if i == 0 else pitch[i - 1]
-            bnds = [pitch0 - 10.0, pitch0 + 10.0]
-            if self.peak_thrust_shaving and found_rated:
-                # Have to constrain thrust
-                const = {}
-                const["type"] = "ineq"
-                const["fun"] = lambda x: constr_Tmax(x, Uhub[i], Omega_rpm[i])
-                params = minimize(
-                    lambda x: maximizePower(x, Uhub[i], Omega_rpm[i]),
-                    pitch0,
-                    method="slsqp",  # "cobyla",
-                    bounds=[bnds],
-                    constraints=const,
-                    tol=TOL,
-                    options={"maxiter": 20},  #'catol':0.01*max_T},
-                )
-                pitch[i] = params.x[0]
-            else:
-                # Only adjust pitch
-                pitch[i] = minimize_scalar(
-                    lambda x: maximizePower(x, Uhub[i], Omega_rpm[i]),
-                    bounds=bnds,
-                    method="bounded",
-                    options={"disp": False, "xatol": TOL, "maxiter": 40},
-                )["x"]
-
-            # Find associated power
-            myout, _ = self.ccblade.evaluate([Uhub[i]], [Omega_rpm[i]], [pitch[i]], coefficients=True)
-            P_aero[i], T[i], Q[i], M[i], Cp_aero[i], Ct_aero[i], Cq_aero[i], Cm_aero[i] = [
-                myout[key] for key in ["P", "T", "Q", "Mb", "CP", "CT", "CQ", "CMb"]
-            ]
-            # P[i], eff[i] = compute_P_and_eff(P_aero[i], P_rated, Omega_rpm[i], driveType, driveEta)
-            eff[i] = np.interp(Omega_rpm[i], lss_rpm, driveEta)
-            P[i] = P_aero[i] * eff[i]
-            Cp[i] = Cp_aero[i] * eff[i]
+        if self.fix_pitch_regI12 == False:
```

Comment:
This seems to apply to all of Region II, not just the Region I.5 where rpm=min_rpm.  Maybe I am misreading?

Reply:
I believe this if statement is a gate that is passed only for regI1/2 or in case of pitch shaving
https://github.com/WISDEM/WISDEM/blob/98952476ad6c3fa3a799851806bf87af801ffd65/wisdem/rotorse/rotor_power.py#L637